### PR TITLE
Use Storybook to publish Jetpack components

### PR DESCRIPTION
**Pull request by gravityrail**
Pulls into #20691

This PR is a proposal for a way to browse and render Jetpack components, both as pure HTML and React.

The goal is to provide developers and designers with a playground for developing and using these components. 

It is based on a [very similar system](https://github.com/WordPress/gutenberg/tree/trunk/storybook) used by Gutenberg (though packaged slightly differently).

You can see the published (example) components at https://automattic.github.io/jetpack/storybook

#### Changes proposed in this Pull Request:
* Add a storybook tool to preview Jetpack UI components.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

To use:

```sh
$ cd projects/js-packages/storybook
$ pnpm install
$ pnpm run storybook:dev
```

This will build and open the Storybook UI in your default browser.

You should see an editor story (under "Playground") as well as a handful of components that already lived in the `@automattic/jetpack-components` directory.

Screenshot:

storybook-browser

### CODE REVIEW

File: .github/workflows/storybook-pages.yml
Comment (kraftbj) [inline .github/workflows/storybook-pages.yml:6]: Should we restrict this to running only on pushes to master that have touched the js-pacakges dir?

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

File: package.json
Comment (kraftbj) [inline package.json:74]: Instead of this, we've had project build commands in the project's `composer.json`. We can do that and be able to run it with `jetpack build js-packages/storybook`

File: projects/js-packages/storybook/composer.json
Comment (kraftbj) [inline projects/js-packages/storybook/composer.json:17]:
```suggestion
		"build-development": "pnpm run storybook:build"
```

**Review by kraftbj — CHANGES_REQUESTED** [review]
A couple of small things.

File: .github/workflows/storybook-pages.yml
Comment (kraftbj) [reply .github/workflows/storybook-pages.yml:6]: 5040b0bff6a34c787a6dd13a88af200d1a612299

File: package.json
Comment (kraftbj) [reply package.json:74]: 932af7aee336a8cfa8fce0dec23e76e822673c7e

File: projects/js-packages/storybook/composer.json
Comment (kraftbj) [reply projects/js-packages/storybook/composer.json:17]: 932af7aee336a8cfa8fce0dec23e76e822673c7e

**Review by kraftbj — DISMISSED** [review]

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
I took out the base-styles addition and the folder renaming. I figure let's add separate things to help make the problematic code easier to spot.

Also, adding base-styles separately, hopefully that'll help speed up other work while figuring out everything that's going on in this PR.

::Please don't merge this until it is reset to master. Updated the base to the branch adding the new base-styles package.::

**Review by kraftbj — APPROVED** [review]

File: projects/plugins/jetpack/extensions/shared/test/block-fixtures.js
Comment (brbrr) [inline projects/plugins/jetpack/extensions/shared/test/block-fixtures.js:100]:
```suggestion
				const validationIssues = gatherValidationIssues( blocksActual );
```

Comment (gravityrail) [reply projects/plugins/jetpack/extensions/shared/test/block-fixtures.js:100]: Thanks but I'm not gonna restart these tests right before merge just to remove two blank lines :) but I will apply this change in my next PR as a cleanup.

Comment (brbrr) [reply projects/plugins/jetpack/extensions/shared/test/block-fixtures.js:100]: I don't see "Ready to Merge" label, that's why I thought it was still under review.

**Review by leogermani — APPROVED** [review]
Approving to unblock the work on other components.

Let's remember to check again after a couple of months to evaluate whether Stroybook is being helpful in our development process or not. If not, we can remove it later... I'll add a reminder for us.

**Comment by kraftbj** [comment]
r230348-wpcom

**Pull request merged**
